apps/frontend/src/components/ui/servers/PanelTerminal.vue

```diff
@@ -1,141 +1,222 @@
-        @click="scrollToBottom"
+        data-pyro-fullscreen
+        :label="isFullScreen ? 'Exit full screen' : 'Enter full screen'"
+        class="experimental-styles-within absolute right-4 top-4 z-[3] grid h-12 w-12 place-content-center rounded-full border-[1px] border-solid border-button-border bg-bg-raised text-contrast transition-all duration-200 hover:scale-110 active:scale-95"
+        @click="toggleFullscreen"
       >
-        <RightArrowIcon class="rotate-90" />
-        <span class="sr-only">Scroll to bottom</span>
+        <LazyUiServersIconsMinimizeIconVue v-if="isFullScreen" />
+        <LazyUiServersIconsFullscreenIcon v-else />
       </button>
-    </Transition>
+
+      <Transition name="fade">
+        <div
+          v-if="hasSelection || isSingleLineSelected"
+          class="absolute right-20 top-4 z-[3] flex flex-row items-center"
+        >
+          <button
+            data-pyro-copy
+            label="Copy selected lines"
+            class="experimental-styles-within flex h-12 flex-row items-center justify-center gap-2 rounded-full border-[1px] border-solid border-button-border bg-bg-raised px-4 text-contrast transition-all duration-200 hover:scale-110 active:scale-95"
+            @click="copySelectedLines"
+          >
+            <CopyIcon class="h-5 w-5" />
+            <span class="">Copy</span>
+          </button>
+          <Transition name="view-button">
+            <button
+              v-if="hasSelection"
+              data-pyro-view
+              label="View full content"
+              class="experimental-styles-within ml-2 flex h-12 w-[100px] flex-row items-center justify-center gap-2 rounded-full border-[1px] border-solid border-button-border bg-bg-raised px-4 text-contrast transition-all duration-200 hover:scale-110 active:scale-95"
+              @click="showSelectedLines"
+            >
+              <EyeIcon class="h-5 w-5" />
+              <span class="">View</span>
+            </button>
+          </Transition>
+        </div>
+      </Transition>
+
+      <Transition name="scroll-to-bottom">
+        <button
+          v-if="bottomThreshold > 0 && !isScrolledToBottom"
+          data-pyro-scrolltobottom
+          label="Scroll to bottom"
+          class="scroll-to-bottom-btn experimental-styles-within absolute bottom-[4.5rem] right-4 z-[3] grid h-12 w-12 place-content-center rounded-full border-[1px] border-solid border-button-border bg-bg-raised text-contrast transition-all duration-200 hover:scale-110 active:scale-95"
+          @click="scrollToBottom"
+        >
+          <RightArrowIcon class="rotate-90" />
+          <span class="sr-only">Scroll to bottom</span>
+        </button>
+      </Transition>
+    </div>
+    <NewModal ref="logModal" class="z-[9999]" header="Viewing full log">
```

Comment:
<img width="707" alt="image" src="https://github.com/user-attachments/assets/0b078261-6e99-413f-986b-d6903cb997bc" />
I selected a few lines and clicked "View", the modal title says "Viewing full log" which seems off